SpatialGDK/Source/SpatialGDK/Private/Interop/SkeletonEntities.cpp

```diff
@@ -0,0 +1,485 @@
+#include "SpatialView/ViewCoordinator.h"
+#include "Utils/ComponentFactory.h"
+#include "Utils/EntityFactory.h"
+#include "Utils/EntityPool.h"
+#include "Utils/InterestFactory.h"
+
+namespace SpatialGDK
+{
+DEFINE_LOG_CATEGORY_STATIC(LogSpatialSkeletonEntityCreator, Log, All);
+
+static ComponentData Convert(const FWorkerComponentData& WorkerComponentData)
+{
+	return ComponentData(OwningComponentDataPtr(WorkerComponentData.schema_type), WorkerComponentData.component_id);
+};
+
+static void ForEachCompleteEntity(bool& bIsFirstCall, const FSubView& SubView, const TFunction<void(Worker_EntityId)> Handler)
+{
+	if (bIsFirstCall)
+	{
+		bIsFirstCall = false;
+		for (const Worker_EntityId EntityId : SubView.GetCompleteEntities())
+		{
+			Handler(EntityId);
+		}
+	}
+	else
+	{
+		for (const EntityDelta& Delta : SubView.GetViewDelta().EntityDeltas)
+		{
+			checkf(Delta.Type != EntityDelta::REMOVE,
+				   TEXT("Skeleton entity manifest must never be removed, but was removed from entity %lld"), Delta.EntityId)
+
+				if (Delta.Type == EntityDelta::ADD || Delta.Type == EntityDelta::UPDATE || Delta.Type == EntityDelta::TEMPORARILY_REMOVED)
+			{
+				const Worker_EntityId EntityId = Delta.EntityId;
+				Handler(EntityId);
+			}
+		}
+	}
+};
+
+FDistributedStartupActorSkeletonEntityCreator::FDistributedStartupActorSkeletonEntityCreator(USpatialNetDriver& InNetDriver)
+	: NetDriver(&InNetDriver)
+	, ManifestSubView(&NetDriver->Connection->GetCoordinator().CreateSubView(
+		  FSkeletonEntityManifest::ComponentId, FSubView::NoFilter,
+		  { NetDriver->Connection->GetCoordinator().CreateComponentChangedRefreshCallback(FSkeletonEntityManifest::ComponentId) }))
+{
+}
+
+void FDistributedStartupActorSkeletonEntityCreator::CreateSkeletonEntitiesForLevel(UWorld& World)
+{
+	ensure(Stage == EStage::Initial);
+
+	UE_LOG(LogSpatialSkeletonEntityCreator, Log, TEXT("Creating skeleton entities for world %s"), *World.GetName())
+
+	Stage = EStage::CreatingEntities;
+
+	TSet<Worker_EntityId_Key> SkeletonEntityIds;
+
+	for (AActor* Actor : TActorRange<AActor>(&World))
+	{
+		if (!IsValid(Actor))
+		{
+			continue;
+		}
+
+		if (!(Actor->IsNameStableForNetworking() && Actor->GetIsReplicated()))
+		{
+			continue;
+		}
+
+		const Worker_EntityId SkeletonEntityId = CreateSkeletonEntityForActor(*Actor);
+
+		UE_LOG(LogSpatialSkeletonEntityCreator, Log, TEXT("EntityId: %lld Creating skeleton entity for %s"), SkeletonEntityId,
+			   *Actor->GetName());
+
+		RemainingSkeletonEntities.Emplace(SkeletonEntityId);
+		SkeletonEntityIds.Emplace(SkeletonEntityId);
+	}
+
+	if (RemainingSkeletonEntities.Num() == 0)
+	{
+		UE_LOG(LogSpatialSkeletonEntityCreator, Log, TEXT("No skeleton entities must be created for %s"), *World.GetName())
+
+		Stage = EStage::DelegatingEntities;
+	}
+}
+
+Worker_EntityId FDistributedStartupActorSkeletonEntityCreator::CreateSkeletonEntityForActor(AActor& Actor)
+{
+	const Worker_EntityId ActorEntityId = NetDriver->PackageMap->AllocateEntityId();
```

Comment:
`ResolveEntityActorAndSubobjects`?

Reply:
Does this resolve the actor? I think at the point you create a skeleton entity you don't actually want to associate the actor to an entity yet.

Reply:
Yes - we've decided not to do this to keep the behavior uniform between the GSM and other workers.